perses/rjmc/topology_proposal.py

```diff
@@ -344,22 +384,145 @@ def _get_mol_atom_map(current_oemol,
+            for scaffold_map in scaffold_maps:
+                if not external_inttypes or allow_ring_breaking:
+                    # reset the IntTypes
+                    for atom in molA.GetAtoms():
+                        atom.SetIntType(0)
+                    for atom in molB.GetAtoms():
+                        atom.SetIntType(0)
+                    index = 1
+                    for scaff_b_id, scaff_a_id in scaffold_map.items():
+                        for atom in molA.GetAtoms():
+                            if atom.GetIdx() == scaffold_A_map[scaff_a_id]:
+                                atom.SetIntType(index)
+                            else:
+                                atom.SetIntType(AtomMapper._assign_atom_ring_id(atom))
+                        for atom in molB.GetAtoms():
+                            if atom.GetIdx() == scaffold_B_map[scaff_b_id]:
+                                atom.SetIntType(index)
+                            else:
+                                atom.SetIntType(AtomMapper._assign_atom_ring_id(atom))
+                        index += 1
+
+                molecule_maps = AtomMapper._get_all_maps(molA, molB,
+                                                     external_inttypes=True,
+                                                     atom_expr=atom_expr,
+                                                     bond_expr=bond_expr)
+                all_molecule_maps.extend(molecule_maps)
+
+        molecule_maps_scores = AtomMapper._remove_rendundant_maps(molA, molB, all_molecule_maps)
+
+        #  TODO - there will be other options that we might want in future here
+        #  maybe _get_mol_atom_map  should return a list of maps and then we have
+        # a pick_map() function elsewhere?
+        # but this would break the API so I'm not doing it now
+        if best == 'geometry':
+            return molecule_maps_scores[min(molecule_maps_scores)]
+        elif best == 'matching_criterion':
+            best_map = AtomMapper._score_nongeometric(molecule_maps_scores.values(), matching_criterion)
+            return best_map
+        elif best == 'random':
+            return np.random.choice(molecule_maps_scores.values())
+        elif best == 'weighted-random':
+            return np.random.choice(molecule_maps_scores.values(),
+                                    [x**-1 for x in molecule_maps_scores.keys()])
+        elif best == 'return-all':
+            _logger.warning('Returning a list of all maps, rather than a dictionary.')
+            return molecule_maps_scores
+
+    @staticmethod
+    def _remove_rendundant_maps(molA, molB, all_molecule_maps):
+        """For a set of maps, it will filter out those that result in
+        the same geometries. From redundant maps, one is chosen randomly.
+
+        Parameters
+        ----------
+        molA : oechem.oemol
+            old molecule
+        molB : oechem.oemol
+            new molecule
+        all_molecule_maps : list(dict)
```

Comment:
Does `all_molecule_maps[index]` contain a dict that maps `molA` atom indices onto `molB` atom indices? Can you give one more line of documentation about what the `dict` structure is?